From bcd3f22607dd659a2884b8cf558dd054c6082d42 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Mon, 11 Jun 2018 17:50:18 +0530
Subject: [PATCH 0363/1851] drm: xlnx: vtc: Use of_match_node()

commit 396816e1c1b7e9602dc899b6fa8f2d27b2bb5de6 from
https://github.com/Xilinx/linux-xlnx.git

Check the given node if it matches the compatible string.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_vtc.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_vtc.c b/drivers/gpu/drm/xlnx/xlnx_vtc.c
index 4776bb290f0c..aa04f7757174 100644
--- a/drivers/gpu/drm/xlnx/xlnx_vtc.c
+++ b/drivers/gpu/drm/xlnx/xlnx_vtc.c
@@ -535,14 +535,26 @@ void xlnx_vtc_disable_vblank_intr(struct xilinx_vtc *vtc)
 	vtc->vblank_fn = NULL;
 }
 
+static const struct of_device_id xilinx_vtc_of_match[] = {
+	{ .compatible = "xlnx,v-tc-5.01.a" },
+	{ /* end of table */ },
+};
+
 /* probe vtc */
 struct xilinx_vtc *xlnx_vtc_probe(struct device *dev,
 				    struct device_node *node)
 {
 	struct xilinx_vtc *vtc;
+	const struct of_device_id *match;
 	struct resource res;
 	int ret;
 
+	match = of_match_node(xilinx_vtc_of_match, node);
+	if (!match) {
+		dev_err(dev, "failed to match the device node\n");
+		return ERR_PTR(-ENODEV);
+	}
+
 	vtc = devm_kzalloc(dev, sizeof(*vtc), GFP_KERNEL);
 	if (!vtc)
 		return ERR_PTR(-ENOMEM);
-- 
2.31.1

